static/js/tools/map/fetcher/breadcrumb_denom_stat.ts

```diff
@@ -20,59 +20,86 @@
 
 import axios from "axios";
 import _ from "lodash";
-import { Dispatch } from "react";
+import { Dispatch, useContext, useEffect } from "react";
 
 import {
   EntitySeriesWrapper,
   SeriesApiResponse,
 } from "../../../shared/stat_types";
 import { stringifyFn } from "../../../utils/axios";
-import { ChartDataType, ChartStoreAction } from "../chart_store";
+import { ChartDataType, ChartStore, ChartStoreAction } from "../chart_store";
+import { useCalculateRatio } from "../compute_hook";
+import { Context } from "../context";
+import { useBreadcrumbDenomStatReady } from "../ready_hook";
 
-export function fetchBreadcrumbDenomStat(
-  entities: string[],
-  statVar: string,
+export function useFetchBreadcrumbDenomStat(
+  chartStore: ChartStore,
   dispatch: Dispatch<ChartStoreAction>
 ): void {
-  const action: ChartStoreAction = {
-    type: ChartDataType.BREADCRUMB_DENOM_STAT,
-    context: {
-      placeInfo: {
-        enclosingPlace: {
-          // entities are the parent places + the enclosing place.
-          // so the last element is the enclosing place.
-          dcid: entities[entities.length - 1],
-          name: "",
+  const { placeInfo, statVar } = useContext(Context);
+  const breadcrumbDenomStatReady = useBreadcrumbDenomStatReady(chartStore);
+  const calculateRatio = useCalculateRatio();
```

Comment:
curious why we're only fetching if calculateRatio is set? since this is just to fetch the data, I think it makes sense to always fetch the data & only do the calculation later if needed

Reply:
This is to save the check in the calling site. Otherwise needs to check it when use this hook. 

The denom stat fetch should only happen when the tool requires compute ratio

Reply:
I feel like we should just fetch any time the stat var and place changes though? Since we won't be blocking on this fetch to complete before rendering, this won't affect the initial rendering time & this way, if user does choose per capita, we don't have to then wait for fetching to complete. What do you think?

Reply:
That's reasonable. Removed this check.